### PR TITLE
Change StringContext scaladoc

### PR DESCRIPTION
**Pull request by ghost**
See @e2a3498 for explanation

### CODE REVIEW

**Comment by ghost** [comment]
Review by @VladUreche

File: src/library/scala/StringContext.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ import scala.annotation.tailrec
  * To provide your own string interpolator, create an implicit class
  * which adds a method to `StringContext`.  Here's an example:
  * {{{
- *    implicit class JsonHelper(val sc: StringContext) extends AnyVal {
+ *    implicit class JsonHelper(sc: StringContext) extends AnyVal {
```
Comment (som-snytt) [inline src/library/scala/StringContext.scala]: The param still needs to be a val; now it can be a private val.

Comment (ghost) [inline src/library/scala/StringContext.scala]: @som-snytt , I thought that `class C(x: R)` logically equivalent to `class C(private[this] val x: R)`. Is it wrong?

Comment (som-snytt) [inline src/library/scala/StringContext.scala]: They are equivalent in the sense that neither emits a field.  But wouldn't want to spoil the adventure of trying the syntax you propose for the example.  Someday, doc examples will be runnable.  Keep the dream alive.

Comment (ghost) [inline src/library/scala/StringContext.scala]: @som-snytt , I agree.

File: src/library/scala/StringContext.scala [hunk 2]
```diff
@@ -38,7 +38,7 @@ import scala.annotation.tailrec
  * To provide your own string interpolator, create an implicit class
  * which adds a method to `StringContext`.  Here's an example:
  * {{{
- *    implicit class JsonHelper(val sc: StringContext) extends AnyVal {
+ *    implicit class JsonHelper(private sc: StringContext) extends AnyVal {
```
Comment (som-snytt) [inline src/library/scala/StringContext.scala]: I guess it will really have to be runnable, so the kitteh can run them.

**Comment by VladUreche** [comment]
Thanks @evvers! Please squash the 3 commits into one, and ping me again so I can merge the PR.

**Comment by ghost** [comment]
@VladUreche , done.

**Pull request merged**